src/representation_theory/group_cohomology_resolution.lean

```diff
@@ -216,10 +223,127 @@ def of_mul_action_basis  :
   basis (fin n → G) (monoid_algebra k G) (of_mul_action k G (fin (n + 1) → G)).as_module :=
 @basis.map _ (monoid_algebra k G) (monoid_algebra k G ⊗[k] ((fin n → G) →₀ k))
   _ _ _ _ _ _ (@algebra.tensor_product.basis k _ (monoid_algebra k G) _ _ ((fin n → G) →₀ k) _ _
-  (fin n → G) (⟨linear_equiv.refl k _⟩)) (of_mul_action_basis_aux k G n)
+  (fin n → G) ⟨linear_equiv.refl k _⟩) (of_mul_action_basis_aux k G n)
 
 lemma of_mul_action_free :
   module.free (monoid_algebra k G) (of_mul_action k G (fin (n + 1) → G)).as_module :=
 module.free.of_basis (of_mul_action_basis k G n)
 
+end basis
+end group_cohomology.resolution
+variables (G)
+
+/-- The simplicial `G`-set sending `[n]` to `Gⁿ⁺¹` equipped with the diagonal action of `G`. -/
+def classifying_space_universal_cover [monoid G] :
+  simplicial_object (Action (Type u) $ Mon.of G) :=
+{ obj := λ n, Action.of_mul_action G (fin (n.unop.len + 1) → G),
+  map := λ m n f,
+  { hom := λ x, x ∘ f.unop.to_order_hom,
+    comm' := λ g, rfl },
+  map_id' := λ n, rfl,
+  map_comp' := λ i j k f g, rfl }
+
+namespace classifying_space_universal_cover
+
+open category_theory.limits category_theory.arrow
+variables (G) [monoid G] (m : simplex_categoryᵒᵖ)
+
+/-- The natural hom of `G`-sets `G ⟶ {pt}` as an arrow. -/
+def arrow : arrow (Action (Type u) (Mon.of G)) :=
+⟨Action.of_mul_action G G, default, ⟨λ x, punit.star, λ g, rfl⟩⟩
+
+/-- The diagram `option (fin (m + 1)) ⥤ G-Set` sending `none` to `{pt}` and `some j` to `G` (with
+the left regular action). -/
+abbreviation wide_cospan :
+  wide_pullback_shape (fin ((opposite.unop m).len + 1)) ⥤ Action (Type u) (Mon.of G) :=
+wide_pullback_shape.wide_cospan default (λ (i : fin (m.unop.len + 1)), Action.of_mul_action G G)
+  (λ i, ⟨λ x, punit.star, λ g, rfl⟩)
+
+/-- The `G-Set` `Gᵐ⁺¹` (with the diagonal action) is the vertex of a cone on `wide_cospan G m`. -/
+def cone : cone (wide_cospan G m) :=
```

Comment:
Renaming `cone` as `wide_cospan.cone` and `is_limit` as `wide_cospan.cone_is_limit` may be helpful.